Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/FileIO/DirectoryManager.cs

```diff
@@ -16,29 +16,34 @@ public class DirectoryManager : IDirectoryManager
         /// </summary>
         public string GetRelativePath(string relativeTo, string path)
         {
-            relativeTo = SanitizePath(relativeTo);
-            path = SanitizePath(path);
+            if (string.IsNullOrEmpty(relativeTo) || string.IsNullOrEmpty(path))
+            {
+                throw new InvalidOperationException($"{nameof(relativeTo)} or {nameof(path)} cannot be null or empty");
+            }
+
+            // If there is no common ancestor, just return the path.
+            if (!string.Equals(SanitizePath(relativeTo).Split('/')[0], SanitizePath(path).Split('/')[0]))
+            {
+                return path;
+            }
+
+            // Calculate the full paths by preserving the OS specific directory separator character and then perform the sanitization.
+            relativeTo = SanitizePath(new DirectoryInfo(relativeTo).FullName);
+            path = SanitizePath(new DirectoryInfo(path).FullName);
 
             var relativeToDirs = relativeTo.Split('/');
             var pathDirs = path.Split('/');
+            var len = Math.Min(relativeToDirs.Length, pathDirs.Length);
 
-            int len = relativeToDirs.Length < pathDirs.Length ? relativeToDirs.Length : pathDirs.Length;
-
-            int lastCommonRoot = -1;
+            var lastCommonRoot = -1;
             int index;
 
             for (index = 0; index < len && string.Equals(relativeToDirs[index], pathDirs[index], StringComparison.OrdinalIgnoreCase); index++)
```

Comment:
On windows using ignoring case is correct but for all other platforms the string comparison should be case sensitive.

Reply:
Fixed